internal/storage/gcs/bucket.go

```diff
@@ -56,6 +89,53 @@ type Bucket interface {
 		ctx context.Context,
 		req *CreateObjectRequest) (*Object, error)
 
+	// CreateChunkUploader creates a new object-chunk-uploader (which implements
```

Comment:
Comments are good in the code, but we should be careful when adding comments.
There is unnecessary and duplicate content in your commetns.

 CreateChunkUploader creates a chunkUploader instance according to supplied createObjectRequest. This needs to be used when caller wants control on the resumable upload api.

This is an alternative to createObject which also uploads data using resumable upload api, but caller has no control on the resumable api.

While creating a chunk-uploader, you can specify a callback called
	// progressFunc which carries (n int64) the total number of bytes
	// successfully uploaded by the uploader so far

The progress function callback is expected exactly once for each upload.

Reply:
Sure, changed to your write-up.